cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1623,8 +1628,8 @@ private String generatePerBrokerConfiguration(NodeRef node, KafkaPool pool, Map<
                                 namespace,
                                 node,
                                 listeners,
-                                listenerId -> advertisedHostnames.get(node.nodeId()).get(listenerId),
-                                listenerId -> advertisedPorts.get(node.nodeId()).get(listenerId)
+                                listenerId -> Optional.ofNullable(advertisedHostnames.get(node.nodeId())).orElse(Map.of()).get(listenerId),
+                                listenerId -> Optional.ofNullable(advertisedPorts.get(node.nodeId())).orElse(Map.of()).get(listenerId)
```

Comment:
Just thinking about the readability of this line ...

AFAIU, you get `advertisedHostnames` as empty map so the `get(node.nodeId())` returns null (so you got the NPE when trying to look for the listenerId). Now, you are using this null to create an nullable `Optional` which because is null, is replaced by a `Map.of()` on which you look for listenerId which of course returns null.
I find this really hard to read/understand.

Why not just a simple:

```java
advertisedHostnames.size() != 0 ? advertisedHostnames.get(node.nodeId()).get(listenerId) : null
```

But again it's my personal preference. Let's see what others think.

Reply:
An alternative would be:
```
listenerId -> Optional.ofNullable(advertisedHostnames.get(node.nodeId()))
        .map(hosts -> hosts.get(listenerId))
        .orElse(null),
```

Reply:
That would also protect us in future if advertisedHostnames has size > 0 but is missing the id we are looking for right?

Reply:
Right, my proposal was missing that with a potential new NPE. Anyway before making any change we should agree on why this is necessary (see discussion on the comment with Tina and Jakub. It's not still clear to me neither.

Reply:
this change is reverted based on the discussion below.